app/models/service_template_ansible_playbook.rb

```diff
@@ -163,4 +165,24 @@ def destroy
     end
     super
   end
+
+  # ServiceTemplate includes a retirement resource action
+  #   with a defined job template:
+  #
+  #   1. A resource_action that includes a configuration_template_id.
+  #   2. At least one service instance where :retired is set to false.
+  #
+  def potential_retirement?
+    retirement_jt_exists = resource_actions.where("action = 'Retirement' AND configuration_template_id IS NOT NULL").present?
```

Comment:
@syncrou you missed the previous comment. The better way can be `resource_actions.where(:action => 'Retirement').where.not(:configuration_template => nil).exists?`